### ISSUE TITLE
Badge is stuck 'pending'

### ISSUE THREAD

**Issue by MikeKovarik** [alanshaw/david-www#419]
I've tried to use david on couple of my projects throughout this year and it only ever worked for a couple of days. I tried to refresh the page and badge couple of times, come back in a day or two, login to david's homepage and all the usual things i could come up with. With no result.

For example I haven't updated [rage-edit](https://github.com/MikeKovarik/rage-edit) for couple of months now and it only has three dev-deps so i don't understand why is it still pending after all this time.

Am I doing something wrong?
Why does David hate me? :D

**Comment by erayhanoglu** [comment]
Me too

**Comment by tr1s** [comment]
Same here. Any word on this being fixed?

**Comment by alanshaw** [comment]
Depending on the queue of badges to create it can take some time. If you have a badge stuck in "pending" then you need to post a link to it so that I can investigate.

**Comment by erayhanoglu** [comment]
Here is some examples. This problem occurs frequently.
https://david-dm.org/panates/putil-promisify?type=dev
https://david-dm.org/panates/putil-waterfall
https://david-dm.org/panates/sqb-serializer-pg
https://david-dm.org/panates/sqb-serializer-pg?type=peer

**Comment by tr1s** [comment]
@alanshaw yeah just took a couple days I guess. Seems fine now!

**Comment by Borewit** [comment]
Duplicate of bug #368.

**Comment by EricLambrecht** [comment]
I have the same problem. Didn't update for 14 days now, yet it's still pending:

[![dependencies Status](https://david-dm.org/ericlambrecht/spotify-magician/status.svg)](https://david-dm.org/ericlambrecht/spotify-magician)
[![devDependencies Status](https://david-dm.org/ericlambrecht/spotify-magician/dev-status.svg)](https://david-dm.org/ericlambrecht/spotify-magician?type=dev)

My [`package.json`](https://github.com/EricLambrecht/spotify-magician/blob/master/package.json)

**Comment by Borewit** [comment]
It looks like the pending badge shields turned into a proper status to me.

**Comment by buskerone** [comment]
this is a recurrent problem

**Comment by YasharF** [comment]
Yes, from what I have captured this is recurring perhaps for when the badges are not called often enough.

https://github.com/pugjs/pug/pull/3146#issuecomment-498301700.  From earlier today:

At 8:28 PDT
![image](https://user-images.githubusercontent.com/6448697/58814407-827b6380-85da-11e9-8acc-7ab0b287386b.png)

At 8:33 PDT
![image](https://user-images.githubusercontent.com/6448697/58814518-bb1b3d00-85da-11e9-95d8-dc54baf43519.png)

**Comment by ilyakam** [comment]
> If you have a badge stuck in "pending" then you need to post a link to it so that I can investigate.

Here's the link: https://david-dm.org/ilyakam/gulp-pug-linter?type=dev

It has been stuck like this for many months (possibly even years) now.

**Comment by ferrao** [comment]
Same thing here: https://david-dm.org/academia-de-codigo/noire-server

**Comment by iansan5653** [comment]
Same here https://github.com/iansan5653/unraw

**Comment by renemadsen** [comment]
I get an Internal Server Error:

[https://david-dm.org/microting/eform-angular-frontend/eform-client/](https://david-dm.org/microting/eform-angular-frontend/eform-client/)

Is it because david-dm does not support package.json being placed in a subfolder?
That would be awesome, since the package.json cannot be moved, since the repository contains mixed content.

Repository in question is https://github.com/microting/eform-angular-frontend

**Comment by YasharF** [comment]
@renemadsen, you would need to use the `path` query parameter if package.json is not in the root. aka `https://david-dm.org/microting/eform-angular-frontend?path=eform-client`

 [![Dependencies Status](https://david-dm.org/microting/eform-angular-frontend/status.svg?path=eform-client)](https://david-dm.org/microting/eform-angular-frontend?path=eform-client) 

 [![Dependencies Status](https://david-dm.org/microting/eform-angular-frontend/dev-status.svg?path=eform-client)](https://david-dm.org/microting/eform-angular-frontend?path=eform-client&type=dev)

**Comment by renemadsen** [comment]
@YasharF Thanks,

Where in the documentation is it possible to find this, maybe I'm not looking the right place.

**Comment by KingDarBoja** [comment]
@YasharF This approach doesn't work for me, had to use for the dev one:
`[![Dependencies Status](https://david-dm.org/org/repo.svg?path=subrepo/path)](https://david-dm.org/org/repo?path=subrepo/path)`

But for the dev ones, I have no idea how to accomplish the same. Your approach only displays the `devDepencency` on pending status too.

**Comment by YasharF** [comment]
@KingDarBoja See https://github.com/alanshaw/david-www/issues/419#issuecomment-498310597

**Comment by KingDarBoja** [comment]
@YasharF So I do have to visit the url for each one in order to trigger the queue so the badges can be updated once again? Seems to be a bit annoying.

**Comment by YasharF** [comment]
@KingDarBoja I am not a maintainer for this package, please don't tag me on postings.

**Comment by plastikfan** [comment]
Not sure what I'm missing, but how do I get my package on david-dm website? I've linked david-dm to my github account, but I don't lnow what to do next. If you go to david-dm.org/plastikfan/zenobia-ts, you just see an internal server error message instead the the page showing the state of dependencies. I also created a github token, but I don't know which rights its needs so I guessed, but its not working. What else do I need to do?

**Comment by alanshaw** [comment]
Add a package.json file to your repo https://github.com/plastikfan/zenobia-ts
